Backend/Backend/Infrastructure/AppDbContext.cs

```diff
@@ -325,6 +327,14 @@ public override int SaveChanges()
 
             ((BaseEntity)entry.Entity).UpdatedAt = currentTime;
         }
+        
+        // Check for soft delete
+        var softDeleteEntries = ChangeTracker.Entries()
```

Comment:
This is completely fine for now, but in the future we might want to leave this up to the database itself to run, and have a job that runs at some time like midnight that cleans up these entities

Reply:
Remind me though, did we make it so that entities that are softdeleted are not searchable anymore either?